api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirEncounterDaoImpl.java

```diff
@@ -50,4 +81,53 @@ protected void setupSearchParams(Criteria criteria, SearchParameterMap theParams
 			}
 		});
 	}
+	
+	private int getMaxParameter(SearchParameterMap theParams) {
```

Comment:
It would maybe be a good idea to try to refactor this and `getTopNRankedUuids()` into utility classes so we can prevent code duplication. Maybe instead of having a `LastnEncountersResults` and `LastnObservationResult` we can get away with a `LastnResult` object which holds the UUID and relevant date then something like a `HashMap` to store other properties (e.g., concept)